From 0d68f8bbba6a4c50c472946630820196f8bf63b6 Mon Sep 17 00:00:00 2001
From: Marek 'marx' Grac <mgrac@redhat.com>
Date: Wed, 18 Feb 2015 14:23:34 +0100
Subject: [PATCH 07/16] Revert "COMPATIBILITY BREAK: Remove obsolete options
 from STDIN"

For historical reasons and backward compatibility we supported these options
        blade -> port
        option -> action
        fm -> port
        hostname -> ipaddr

These options were not in documentation and are completely removed now.
---
 fence/agents/lib/fencing.py.py | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/fence/agents/lib/fencing.py.py b/fence/agents/lib/fencing.py.py
index c99aa91..ec5f855 100644
--- a/fence/agents/lib/fencing.py.py
+++ b/fence/agents/lib/fencing.py.py
@@ -1244,6 +1244,16 @@ def _parse_input_stdin(avail_opt):
 		(name, value) = (line + "=").split("=", 1)
 		value = value[:-1]
 
+		# Backward compatibility (RHEL6 only)
+		if name == "blade":
+			name = "port"
+		elif name == "option":
+			name = "action"
+		elif name == "fm":
+			name = "port"
+		elif name == "hostname":
+			name = "ipaddr"
+
 		if avail_opt.count(name) == 0 and name in ["nodename"]:
 			continue
 		elif avail_opt.count(name) == 0:
-- 
1.9.3

